profile/store.go

```diff
@@ -24,6 +25,8 @@ type Store interface {
 	Owner() *Profile
 	// SetOwner handles updates to the current user profile at runtime
 	SetOwner(own *Profile) error
+	// profile stores are built atop a store of keypairs
+	KeyStore() key.Store
```

Comment:
I'm not sure we want to surface the KeyStore in the interface. This starts to recreate the "repo problem" where we have different functions relying on different sources for interface accessors.

Obvi the store constructor / internal state will still need a keystore tho